stream/src/main/scala/libretto/stream/scaletto/BinarySearchTree.scala

```diff
@@ -428,26 +428,25 @@ class BinarySearchTree[DSL <: Scaletto, CLib <: CoreLib[DSL], SLib <: ScalettoLi
   }
 
   private object Absorptive {
-    implicit def fromTransportive[F[_]](using F: Transportive[F]): Absorptive[F] =
-      new Absorptive[F] {
-        override def lift[A, B](f: A -⚬ B): F[A] -⚬ F[B] =
-          F.lift(f)
+    given [F[_]](using F: Transportive[F]): Absorptive[F] with {
+      override def lift[A, B](f: A -⚬ B): F[A] -⚬ F[B] =
+        F.lift(f)
 
-        override def absorbOrNeglectL[A, B](using CloseableCosemigroup[A]): (A |*| F[B]) -⚬ F[A |*| B] =
-          F.inL
+      override def absorbOrNeglectL[A, B](using CloseableCosemigroup[A]): (A |*| F[B]) -⚬ F[A |*| B] =
+        F.inL
 
-        override def absorbOrNeglectR[A, B](using CloseableCosemigroup[B]): (F[A] |*| B) -⚬ F[A |*| B] =
-          F.inR
+      override def absorbOrNeglectR[A, B](using CloseableCosemigroup[B]): (F[A] |*| B) -⚬ F[A |*| B] =
+        F.inR
 
-        override def absorbL[A, B, C](combine: (A |*| B) -⚬ C, recover: (A |*| Done) -⚬ C): (A |*| F[B]) -⚬ F[C] =
-          F.inL > F.lift(combine)
+      override def absorbL[A, B, C](combine: (A |*| B) -⚬ C, recover: (A |*| Done) -⚬ C): (A |*| F[B]) -⚬ F[C] =
+        F.inL > F.lift(combine)
 
-        override def absorbR[A, B, C](combine: (A |*| B) -⚬ C, recover: (Done |*| B) -⚬ C): (F[A] |*| B) -⚬ F[C] =
-          F.inR > F.lift(combine)
-      }
+      override def absorbR[A, B, C](combine: (A |*| B) -⚬ C, recover: (Done |*| B) -⚬ C): (F[A] |*| B) -⚬ F[C] =
+        F.inR > F.lift(combine)
+    }
 
-    implicit def insideTransportive[F[_], G[_]](using F: Transportive[F], G: Absorptive[G]): Absorptive[λ[x => F[G[x]]]] =
-      G >>> F
+    given insideTransportive[F[_], G[_]](using F: Transportive[F], G: Absorptive[G]): Absorptive[λ[x => F[G[x]]]] =
+      G .>>> (using F)
```

Comment:
Well, could be better…

(But didn't like to rewrite anything substantial without discussion)

Reply:
I guess `>>>` should just take an explicit (non-given) parameter.